### DISCUSSION TITLE
pdb refreshing restarting

### DISCUSSION THREAD

**Discussion by saloni-2404** [strimzi/strimzi-kafka-operator#5007] [General]
kafka-365dayssl-kafka             N/A             1                 1                     16s
kafka-365dayssl-zookeeper         N/A             1                 1                     18s
kafka-blackbox-kafka              N/A             1                 1                     16s
kafka-blackbox-zookeeper          N/A             1                 1                     18s

It keeps restarting as in the age, it keeps refreshing every 60/120 seconds. Is this expected behaviour?

**Comment by saloni-2404** [comment]
Also, when we try setting the value of maxUnavailable: 3(in a 5replica cluster), it still shows up as:
NAME                   MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
my-cluster-zookeeper   N/A             1                 1                     41s

**Reply by scholzj** [reply]
What Strimzi version are you using? How are you configuring it?

**Reply by saloni-2404** [reply]
Hi Jakub, it is 0.17.0

kafka.template.pod:
        podDisruptionBudget:
          maxUnavailable: 3

**Reply by scholzj** [reply]
So, the refreshing of the PDB every two minutes is something we fixed at some point - I assume after 0.17. So I think Strimzi upgrade would solve that.

You did not really shared the whole YAML, so it is not clear what exactly you have there. But the `podDisruptionBudget` is supposed to be directly under `.spec.kafka.template`. It is not part of the `pod` template.

**Reply by saloni-2404** [reply]
Okay, Jakub, and it would be the same for 0.20.0 version as well? under .spec.kafka.template, correct?

**Reply by scholzj** [reply]
The PDB configuration is still in the same place. The patching was fixed in 0.20 in #3640.

**Reply by saloni-2404** [reply]
ok thanks Jakub!
